File: frontend/lib/models/user_interaction.dart
class UserInteraction {
  final int id;
  final int userId;
  final int kosId;
  final int? rating;
  final bool isFavorite;
  final int clickCount;

  UserInteraction({
    required this.id,
    required this.userId,
    required this.kosId,
    this.rating,
    required this.isFavorite,
    required this.clickCount,
  });

  factory UserInteraction.fromJson(Map<String, dynamic> json) {
    return UserInteraction(
      id: json['id'] ?? 0,
      userId: json['user_id'] ?? 0,
      kosId: json['kos_id'] ?? 0,
      rating: json['rating'],
      isFavorite: (json['is_favorite'] == 1 || json['is_favorite'] == true || json['is_favorite'] == '1'),
      clickCount: json['click_count'] ?? 0,
    );
  }
}

File: backend/app/Models/UserInteraction.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class UserInteraction extends Model
{
    protected $fillable = [
        'user_id',
        'kos_id',
        'rating',
        'is_favorite',
        'click_count',
        'favorited_price_snapshot',
        'favorited_rooms_snapshot',
    ];

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function kos()
    {
        return $this->belongsTo(Kos::class);
    }
}
